CMakeLists.txt

```diff
@@ -67,6 +67,22 @@ else()
   find_program(GO_EXECUTABLE go)
 endif()
 
+if(AWS_LC_BUILD_WITHOUT_PERL)
+  message(STATUS "AWS_LC_BUILD_WITHOUT_PERL is enabled.")
+  if(PERL_EXECUTABLE)
+    message(STATUS "Unset PERL_EXECUTABLE: ${PERL_EXECUTABLE}.")
+    unset(PERL_EXECUTABLE CACHE)
+  endif()
+endif()
+
+if(AWS_LC_BUILD_WITHOUT_GO)
+  message(STATUS "AWS_LC_BUILD_WITHOUT_GO is enabled.")
+  if(GO_EXECUTABLE)
+    message(STATUS "Unset GO_EXECUTABLE: ${GO_EXECUTABLE}.")
+    unset(GO_EXECUTABLE CACHE)
+  endif()
+endif()
+
```

Comment:
I'm not entirely sure that I like this change. While we do provide consumers the ability to build without these dependencies, I don't know if it makes sense to provide them a way to opt-out of using them even if they do have them installed.

Are we not able to just remove these dependencies from our Dockerfiles to test this?

Reply:
These build flags can help test when the system has `GO/Perl` installed. So developers don't need to switch development env.  
All Docker images used by `aws-c-cal` CI have `Perl` installed. I tried some uninstall commands but failed to remove `Perl`.

Reply:
I also talked with @andrewhop a bit about this and this change makes sense to be more explicit by providing these flags, or auto-detect whether these dependencies are not installed (which is the current behavior). I just think that we should do one of these and not both.